packages/ui/src/components/XPath/XPathEditorLayout.tsx

```diff
@@ -69,23 +78,45 @@ export const XPathEditorLayout: FunctionComponent<XPathEditorLayoutProps> = ({ m
               <TabContent id="functions" className="xpath-editor-layout-tab-content">
                 <Menu isScrollable className="xpath-editor-layout-tab-content">
                   <MenuContent menuHeight="100%" maxMenuHeight="100%">
+                    <MenuItem data-testid="functions-menu-search-item" onFocus={focusOnSearchInput}>
+                      <SearchInput
+                        data-testid="functions-menu-search-input"
+                        ref={searchInputRef}
+                        placeholder="Filter functions..."
+                        value={searchValue}
+                        onChange={handleOnSearchChange}
+                      />
+                    </MenuItem>
+
                     {Object.keys(functionDefinitions).map((value) => (
-                      <MenuGroup key={value} label={value}>
-                        {functionDefinitions[value as FunctionGroup].map((func, index) => (
-                          <DraggableContainer
-                            key={index}
-                            id={`${value}-${index}-${func.name}`}
-                            nodeData={new FunctionNodeData(func)}
-                          >
-                            <MenuItem
-                              className="menu-item-drag"
-                              key={`${value}-${index}`}
-                              description={func.description}
+                      <MenuGroup
+                        key={value}
+                        label={value}
+                        hidden={
+                          !functionDefinitions[value as FunctionGroup].some((func) =>
+                            func.displayName.toLocaleLowerCase().includes(searchValue.toLocaleLowerCase()),
```

Comment:
Could we transform the searchValue to lowercase once and then use it? This way we would save some work

Reply:
I checked this and doesn't seem to work as expected - the search input would be transformed to lowercase, which is not something we want. So leaving as is now is I think better option.